### PR TITLE
feat: SectionList support (without stickyHeaders for now)

### PR DESCRIPTION
**Pull request by alexco2**
Added SectionList Support. Also updated documentation and added example tab with a SectionList.
As we know from https://github.com/PedroBern/react-native-collapsible-tab-view/issues/136 stickyHeaders cause a problem. This issue also applies to SectionList. I played around with paddingTop, but couldn´t solve the issue.
I also upgraded Expo to v42 and reanimated to v2.2.2.

### CODE REVIEW

**Comment by alexco2** [comment]
I initially wanted to create a seperate pull request for the reanimated and Expo update, but it looks like I added the commit to this pull request. Sorry, I´m still a bit new to git. Anyway, this pr would close this issue: https://github.com/PedroBern/react-native-collapsible-tab-view/issues/164

**Comment by PedroBern** [comment]
hi @alexco2 thanks for the PR, appreciate the bump on expo and reanimated 🙌  But it has a bug (don't know if there are more bugs, I stopped testing after this due to time constraints on my end):

https://user-images.githubusercontent.com/39778068/135697171-0e2758be-5ef3-4149-be7f-a14da398363d.mp4

**Comment by alexco2** [comment]
> Added SectionList Support. Also updated documentation and added example tab with a SectionList.
> As we know from https://github.com/PedroBern/react-native-collapsible-tab-view/issues/136 stickyHeaders cause a problem. This issue also applies to SectionList. I played around with paddingTop, but couldn´t solve the issue.
> I also upgraded Expo to v42 and reanimated to v2.2.2.

Hi @PedroBern 
Thank you for the quick reply. I've also noticed this bug. It's not a new one since it is the same one as in https://github.com/PedroBern/react-native-collapsible-tab-view/issues/136. 
I tried fixing it but wasn't successful. Should it be disabled by default on iOS?

File: example/src/Shared/SectionContacts.tsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      renderSectionHeader={({ section: { title } }) => (
+        <Text style={styles.header}>{title}</Text>
+      )}
+      ItemSeparatorComponent={ItemSeparator}
+      ListEmptyComponent={ListEmptyComponent}
+      // see https://github.com/software-mansion/react-native-reanimated/issues/1703
+      onRefresh={Platform.OS === 'ios' ? startRefreshing : undefined}
+      refreshing={Platform.OS === 'ios' ? isRefreshing : undefined}
+      nestedScrollEnabled={nestedScrollEnabled}
+    />
```
Comment (PedroBern) [inline example/src/Shared/SectionContacts.tsx:160]: add `stickySectionHeadersEnabled={false}` please to avoid the #136 bug

**Review by PedroBern — CHANGES_REQUESTED** [review]

**Comment by PedroBern** [comment]
@alexco2 with the required review changes looks good to me 🚀  thanks for the PR 🎉 

@andreialecu could you please review as well?

File: example/src/Shared/SectionContacts.tsx (hunk 1 above)
Comment (alexco2) [reply example/src/Shared/SectionContacts.tsx]: done

**Review by PedroBern — APPROVED** [review]

**Comment by andreialecu** [comment]
I'll take a look tomorrow.

**Comment by andreialecu** [comment]
Released in 4.3.0 🚀

**Comment by alexco2** [comment]
@PedroBern @andreialecu 
Amazing. This is my first PR 🎉. Thank you for merging and releasing so fast. Really motivates me to commit further to this library!

**Comment by andreialecu** [comment]
Thank you @alexco2, keep 'em coming! 💯

**Pull request merged**